pkg/manager/member/pd_upgrader.go

```diff
@@ -146,6 +140,68 @@ func (u *pdUpgrader) transferPDLeaderTo(tc *v1alpha1.TidbCluster, targetName str
 	return controller.GetPDClient(u.deps.PDControl, tc).TransferPDLeader(targetName)
 }
 
+// choosePDToTransferFromMembers choose a pd to transfer leader from members
+//
+// Assume that current leader ordinal is x, and range is [0, n]
+//	1. Find the max suitable ordinal in (x, n], because they have been upgraded
+//	2. If no suitable ordinal, find the min suitable ordinal in [0, x) to reduce the count of transfer
+func choosePDToTransferFromMembers(tc *v1alpha1.TidbCluster, newSet *apps.StatefulSet, ordinal int32) string {
+	tcName := tc.GetName()
+	ordinals := helper.GetPodOrdinals(*newSet.Spec.Replicas, newSet)
+
+	genPDName := func(targetOrdinal int32) string {
+		pdName := PdName(tcName, targetOrdinal, tc.Namespace, tc.Spec.ClusterDomain)
+		if _, exist := tc.Status.PD.Members[pdName]; !exist {
+			pdName = PdPodName(tcName, targetOrdinal)
+		}
+		return pdName
+	}
+	pred := func(pdName string) bool {
+		return tc.Status.PD.Members[pdName].Health
+	}
+
+	// set ordinal to max ordinal if ordinal isn't exist
+	if !ordinals.Has(ordinal) {
```

Comment:
What's the case here? Does it mean a Pod that is not desired is being upgraded?

Reply:
it's mean that current sts don't contain the pod, and regard the pod as the max ordinal, because all pods in sts haven't been upgraded